File: app/src/main/java/com/zcorp/opensportmanagement/data/pref/IPreferencesHelper.java
package com.zcorp.opensportmanagement.data.pref;

import com.zcorp.opensportmanagement.data.IDataManager;

/**
 * Created by janisharali on 27/01/17.
 */

public interface IPreferencesHelper {

    int getCurrentUserLoggedInMode();

    void setCurrentUserLoggedInMode(IDataManager.LoggedInMode mode);

    Long getCurrentUserId();

    void setCurrentUserId(Long userId);

    String getCurrentUserName();

    void setCurrentUserName(String userName);

    String getCurrentUserEmail();

    void setCurrentUserEmail(String email);

    String getCurrentUserProfilePicUrl();

    void setCurrentUserProfilePicUrl(String profilePicUrl);

    String getAccessToken();

    void setAccessToken(String accessToken);

}

File: app/src/main/java/com/zcorp/opensportmanagement/data/pref/IPreferencesHelper.kt
package com.zcorp.opensportmanagement.data.pref

/**
 * Created by romainz on 16/02/18.
 */
interface IPreferencesHelper {
    fun getCurrentUserName(): String
    fun setCurrentUserName(username: String)
    fun getCurrentUserEmail(): String
    fun setCurrentUserEmail(email: String)
    fun getCurrentUserProfilePicUrl(): String
    fun setCurrentUserProfilePicUrl(profilePicUrl: String)
    fun getAccessToken(): String
    fun setAccessToken(accessToken: String)
    fun getCurrentTeamId(): Int
    fun setCurrentTeamId(teamId: Int)
    fun setAvailableTeamIds(availableTeams: List<Int>)
    fun getAvailableTeamIds(): List<Int>
}